src/main/scala/intellij/haskell/editor/HaskellCompletionContributor.scala

```diff
@@ -182,6 +184,10 @@ class HaskellCompletionContributor extends CompletionContributor {
                 resultSet.addAllElements(findTopLevelTypeSignatureLookupElements(psiFile).asJavaCollection)
               }
 
+              if (HaskellConsoleView.isConsole(psiFile)) {
```

Comment:
We have to make separation in code between console file and normal file. Otherwise everything get mixed up and not maintainalbe anymore.

Also there is BrowseModuleComponent which has cached the top level identifiers for all modules.